src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -27,6 +30,14 @@ public class DependencyGraphSpec
         private const int Version = 1;
 
         private readonly bool _isReadOnly;
+        // Internal for testing purposes
+        internal Dictionary<string, string>? _projectNameToHashCode;
```

Comment:
```suggestion
        internal Dictionary<string, string>? _projectNameToHashCodeCache;
```

Or Lookup?  The name feels confusing to me at the moment. The name suggests it should be a single value, not a collection